gym/wrappers/one_hot_observation.py

```diff
@@ -0,0 +1,43 @@
+import gym
+import numpy as np
+
+
+def one_hot(a: np.array, size: int) -> np.array:
+    b = np.zeros((size), dtype=np.float32)
+    b[a] = 1
+    return b
+
+
+class OneHotObservation(gym.ObservationWrapper):
+    """
+    This wrapper one hot encodes the `Discrete` observation space.
+    For example, `Discrete(3)` observation space will be converted
+    to `Box(0, 1, (3,))` observation space, where
+    `(1, 0, 0)`, `(0, 1, 0)` and `(0, 0, 1)` corresponds to the three
+    discrete states
+
+    Example::
+
+        >>> import gym
+        >>> env = gym.make('Taxi-v3')
+        >>> env = gym.wrappers.OneHotObservation(env)
+        >>> env.reset()
+        >>> env.observation_space
+        Box(0, 1, (500,))
+
+    Args:
+        env (Env): environment
+    """
+
+    def __init__(self, env: gym.Env):
+        super().__init__(env)
+        self.n: int = None
```

Comment:
I don't think this actually does anything? It will either be overwritten a few lines down, or there will be an exception